analyzer/src/main/kotlin/managers/Cargo.kt

```diff
@@ -251,39 +251,42 @@ class Cargo(
             .single { it.startsWith("$projectName $projectVersion") }
 
         val projectNode = metadata["packages"].single { it["id"].textValueOrEmpty() == projectId }
+        val groupedProjectDependencies = projectNode["dependencies"].groupBy { it["kind"].textValueOrEmpty() }
 
-        fun filterDependencies(condition: (String) -> Boolean) =
-            projectNode["dependencies"].filter { node ->
-                val kind = node["kind"].textValueOrEmpty()
-                condition(kind)
-            }.mapNotNull {
-                val dependencyName = it["name"].textValue()
-                val version = getResolvedVersion(projectName, projectVersion, dependencyName, metadata)
-                if (version == null) null else Pair(dependencyName, version)
-            }
+        val nameToVersionMapping = { dependency: JsonNode ->
+            val dependencyName = dependency["name"].textValue()
+            val version = getResolvedVersion(projectName, projectVersion, dependencyName, metadata)
+            version?.let { Pair(dependencyName, it) }
+        }
 
-        val directDependencies = filterDependencies { kind -> kind != "dev" && kind != "build" }
-        val directDevDependencies = filterDependencies { kind -> kind == "dev" }
-        val directBuildDependencies = filterDependencies { kind -> kind == "build" }
+        val dependencies = groupedProjectDependencies[""]
+            .orEmpty()
+            .mapNotNull(nameToVersionMapping)
+            .map { buildDependencyTree(name = it.first, version = it.second, packages = packages, metadata = metadata) }
+            .toSortedSet()
```

Comment:
The four lines above are repeated three times, can you move this to a helper function? And maybe combine it with creating the scope for even less repetition?

Reply:
Most of that repetition was already preset before my change, but done anyway!